tests/sentry/tasks/test_assemble.py

```diff
@@ -756,16 +757,39 @@ def _create_bundle_and_bind_to_release(self, release, dist, bundle_id, indexing_
 
         return artifact_bundle
 
+    def mock_assemble_result(self) -> AssembleResult:
+        bundle_file = self.create_artifact_bundle_zip(
+            fixture_path="artifact_bundle_debug_ids", project=self.project.id
+        )
+        blob1 = FileBlob.from_file(ContentFile(bundle_file))
+        total_checksum = sha1(bundle_file).hexdigest()
+        return assemble_file(
+            task=AssembleTask.ARTIFACT_BUNDLE,
+            org_or_project=self.organization,
+            name="bundle.zip",
+            checksum=total_checksum,
+            chunks=[blob1.checksum],
+            file_type="artifact.bundle",
+        )
+
     @patch("sentry.tasks.assemble.index_artifact_bundles_for_release")
     def test_index_if_needed_with_no_bundles(self, index_artifact_bundles_for_release):
         release = "1.0"
         dist = "android"
 
-        _index_bundle_if_needed(
-            org_id=self.organization.id, release=release, dist=dist, date_snapshot=datetime.now()
+        post_assembler = ArtifactBundlePostAssembler(
+            assemble_result=self.mock_assemble_result(),
+            organization=self.organization,
+            release=release,
+            dist=dist,
+            project_ids=[],
+        )
+        post_assembler._index_bundle_if_needed(
+            release=release, dist=dist, date_snapshot=datetime.now()
         )
 
         index_artifact_bundles_for_release.assert_not_called()
+        post_assembler.close()
```

Comment:
You should maybe use `with` as this is a scope guard. Also the constructor is always the same, so maybe a `self` method could reduce copy-pasta here.